Software/Program/Code/Core/Inc/main.h

```diff
@@ -31,22 +31,60 @@ extern "C" {
 
 /* Private includes ----------------------------------------------------------*/
 /* USER CODE BEGIN Includes */
-
+#include <stdbool.h>
+#include <stdint.h>
 /* USER CODE END Includes */
 
 /* Exported types ------------------------------------------------------------*/
 /* USER CODE BEGIN ET */
 
+enum Direction_E
+{
+   DIRECTION_FORWARD_MIN,
+   DIRECTION_FORWARD_NOMINAL,
+   DIRECTION_FORWARD_MAX,
+   DIRECTION_REVERSE_MIN,
+   DIRECTION_REVERSE_NOMINAL,
+   DIRECTION_REVERSE_MAX,
+   DIRECTION_ERROR,
+   DIRECTION_NOT_AVAILABLE
+};
+
+typedef float    Hz_T;
+typedef uint32_t us_T;
```

Comment:
Does not need to be shared. In fact, create a separate module that takes as input the user setpoints and outputs the necessary square wave timer-based toggle points. Much of the modifications to this file will have this comment applied to them.